src/utils/providers/ReadonlyRpcProvider.ts

```diff
@@ -0,0 +1,23 @@
+import { JsonRpcProvider } from 'ethers'
+import { numberToHex } from '../hex'
+
+class ReadonlyRpcProvider extends JsonRpcProvider {
+  #chainId: string
```

Comment:
Let's not use the `#` syntax and just mark it as private. It's too far off from regular JS.